corehq/apps/locations/models.py

```diff
@@ -680,7 +680,11 @@ def save(self, *args, **kwargs):
         kwargs['sync_to_sql'] = False  # only sync here
         super(Location, self).save(*args, **kwargs)
         if sync_to_sql:
-            self._migration_do_sync()
+            try:
+                self._migration_do_sync()
+            except Exception:
+                self.delete()  # roll back on failure to sync
```

Comment:
ok, yeah that would work as long as there's just 1 save of the couch doc and it's the last operation in the transaction.  i think you'd also have alter the sync to create a couch doc with that id if it doesn't exist